src/System.Linq/src/System/Linq/Single.cs

```diff
@@ -149,5 +150,79 @@ public static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source,
 
             return default(TSource);
         }
+
+        public static bool TrySingle<TSource>(this IEnumerable<TSource> source, out TSource element)
+        {
+            if (source == null)
```

Comment:
Single can throw `Error.NoElements()` or `Error.MoreThanOneElement()` exception. It is `InvalidOperationException` with different messages. 
When TrySingle return false, I did not know what exception `Error.NoElements()` or `Error.MoreThanOneElement()` should be.
I can de-duplicate this code only if add universal text message. I do not  know can I made such changes or can`t.
May be somebody later de-duplicate this code because I know english only a little.